### ISSUE TITLE
Crash on latest version of main

### ISSUE THREAD

**Issue by paulsUsername** [mapbox/mapbox-maps-ios#2119]
NOTE: Not an issue on production. Just affecting at the very least commit: 5e8503c42216a43091bc5abb2f7ab291fcd5c468 (current head as of this posting)

*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Delegate must respond to locationManager:didFailWithError:'
*** First throw call stack:
(0x1b126c870 0x1a9587c00 0x1b07d6d18 0x1be2b9390 0x101fc2390 0x1be2c8774 0x1be2c54ec 0x1be2a21a8 0x1cfb20100 0x1b11b73a8 0x1b11b5b9c 0x1b11b40d4 0x1b11b3668 0x1b11b33cc 0x101fc55d0 0x1b01d1184 0x21abde4d4 0x21abdda10)
libc++abi: terminating due to uncaught exception of type NSException

This happens when you use an instance of CLLocaitonManager to monitor location changes.

Sample code:

```
import Foundation
import UIKit
import MapboxMaps

class TestRecordingViewController: UIViewController {
    var mapView: MapboxMaps.MapView!

    var polylineAnnotaitonManager: PolylineAnnotationManager?

    let appleLocationProvider = AppleLocationProvider()

    var locationObserver: MapboxMaps.Cancelable?
    
    private lazy var locationManager: CLLocationManager = {
        let manager = CLLocationManager()
        manager.delegate = self
        manager.requestWhenInUseAuthorization()
        manager.activityType = .fitness
        manager.pausesLocationUpdatesAutomatically = false
        manager.desiredAccuracy = kCLLocationAccuracyBestForNavigation
        manager.distanceFilter = 25 // meters
        manager.headingFilter = 3 // degrees (1 is default)
        manager.allowsBackgroundLocationUpdates = true
        return manager
    }()
    
    lazy var button: UIButton = {
        let button = UIButton(frame: .init(origin: .init(x: 20, y: 50), size: .init(width: 100, height: 100)))
        button.setTitle("Location", for: .normal)
        button.backgroundColor = .red
        button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
        return button
    }()
    
    var locations: [CLLocation] = []
    
    var locationTrackingCancellation: Cancelable?

    override func viewDidLoad() {
        super.viewDidLoad()
        addMap()
        locationObserver = self.mapView.mapboxMap.onMapLoaded.observe { _ in
            DispatchQueue.main.async {
                self.mapView.location.override(provider: self.appleLocationProvider)
                self.polylineAnnotaitonManager = self.mapView.annotations.makePolylineAnnotationManager()
                self.locationManager.startUpdatingLocation()
            }
        }
        self.view.addSubview(button)

    }
    
    func addMap() {
        mapView = MapView(frame: view.frame)
        mapView.translatesAutoresizingMaskIntoConstraints = false
        if let existingMapView = view.subviews.first(where: { view in
            view is MapView
        }) {
            existingMapView.removeFromSuperview()
        }
        view.insertSubview(mapView, at: 0)
        NSLayoutConstraint.activate([
            mapView.topAnchor.constraint(equalTo: view.topAnchor),
            mapView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
            mapView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            mapView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
        ])
    }
    
    @objc func buttonTapped() {
        self.mapView?.location.options.puckType = .puck2D(.makeDefault())
        self.mapView?.location.options.puckBearing = .heading

        // Enable the heading gesture
        self.mapView?.gestures.options.rotateEnabled = true

        self.locationTrackingCancellation = mapView?.location.onLocationChange.observe { [weak mapView] newLocation in
            guard let location = newLocation.last, let mapView else { return }
            mapView.camera.ease(
                to: CameraOptions(center: location.coordinate, zoom: 15),
                duration: 0.5
            )
        }
    }
}

extension TestRecordingViewController: CLLocationManagerDelegate {
    
    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
        self.locations.append(locations.first!)
        let lineString = LineString(self.locations.map { $0.coordinate })
        var polylineAnnotation = PolylineAnnotation(lineString: lineString)
        polylineAnnotation.lineWidth = 5
        polylineAnnotation.lineColor = StyleColor(UIColor.red)
        self.polylineAnnotaitonManager?.annotations = [polylineAnnotation]
    }

}
```

**Comment by paulsUsername** [comment]
Update, this is also happening on the [v11.2.0-beta.1](https://github.com/mapbox/mapbox-maps-ios/releases/tag/v11.2.0-beta.1).

**Comment by persidskiy** [comment]
@paulsUsername Hi, if you try to implement the `locationManager:didFailWithError:` in `TestRecordingViewController`, will this help?

**Comment by paulsUsername** [comment]
No same issue even with that included.

Exact error:

com.mapbox.common.location.sharedRunloop (17): "Delegate must respond to locationManager:didFailWithError:"

Xcode stack trace:

<img width="375" alt="Screenshot 2024-02-12 at 20 42 00" src="https://github.com/mapbox/mapbox-maps-ios/assets/32492218/a4257a0a-ab42-4729-aae6-fe97151bd27a">

```
import Foundation
import UIKit
import MapboxMaps

class TestRecordingViewController: UIViewController {
    var mapView: MapboxMaps.MapView!

    var polylineAnnotaitonManager: PolylineAnnotationManager?

    let appleLocationProvider = AppleLocationProvider()

    var locationObserver: MapboxMaps.Cancelable?
    
    private lazy var locationManager: CLLocationManager = {
        let manager = CLLocationManager()
        manager.delegate = self
        manager.requestWhenInUseAuthorization()
        manager.activityType = .fitness
        manager.pausesLocationUpdatesAutomatically = false
        manager.desiredAccuracy = kCLLocationAccuracyBestForNavigation
        manager.distanceFilter = 25 // meters
        manager.headingFilter = 3 // degrees (1 is default)
        manager.allowsBackgroundLocationUpdates = true
        return manager
    }()
    
    lazy var button: UIButton = {
        let button = UIButton(frame: .init(origin: .init(x: 20, y: 50), size: .init(width: 100, height: 100)))
        button.setTitle("Location", for: .normal)
        button.backgroundColor = .red
        button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
        return button
    }()
    
    var locations: [CLLocation] = []
    
    var locationTrackingCancellation: Cancelable?

    override func viewDidLoad() {
        super.viewDidLoad()
        addMap()
        locationObserver = self.mapView.mapboxMap.onMapLoaded.observe { _ in
            DispatchQueue.main.async {
                self.mapView.location.override(provider: self.appleLocationProvider)
                self.polylineAnnotaitonManager = self.mapView.annotations.makePolylineAnnotationManager()
                self.locationManager.startUpdatingLocation()
            }
        }
        self.view.addSubview(button)

    }
    
    func addMap() {
        mapView = MapView(frame: view.frame)
        mapView.translatesAutoresizingMaskIntoConstraints = false
        if let existingMapView = view.subviews.first(where: { view in
            view is MapView
        }) {
            existingMapView.removeFromSuperview()
        }
        view.insertSubview(mapView, at: 0)
        NSLayoutConstraint.activate([
            mapView.topAnchor.constraint(equalTo: view.topAnchor),
            mapView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
            mapView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
            mapView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
        ])
    }
    
    @objc func buttonTapped() {
        self.mapView?.location.options.puckType = .puck2D(.makeDefault())
        self.mapView?.location.options.puckBearing = .heading

        // Enable the heading gesture
        self.mapView?.gestures.options.rotateEnabled = true

        self.locationTrackingCancellation = mapView?.location.onLocationChange.observe { [weak mapView] newLocation in
            guard let location = newLocation.last, let mapView else { return }
            mapView.camera.ease(
                to: CameraOptions(center: location.coordinate, zoom: 15),
                duration: 0.5
            )
        }
    }
}

extension TestRecordingViewController: CLLocationManagerDelegate {
    
    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
        self.locations.append(locations.first!)
        let lineString = LineString(self.locations.map { $0.coordinate })
        var polylineAnnotation = PolylineAnnotation(lineString: lineString)
        polylineAnnotation.lineWidth = 5
        polylineAnnotation.lineColor = StyleColor(UIColor.red)
        self.polylineAnnotaitonManager?.annotations = [polylineAnnotation]
    }
    
    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
        print("Error: \(error)")
    }

}
```

**Comment by persidskiy** [comment]
@paulsUsername I got it, thank you for the report! This will be fixed in the upcoming stable release, not in the upcoming rc.

Internal ticket for tracking: https://mapbox.atlassian.net/browse/CORESDK-2328

**Comment by aleksproger** [comment]
@paulsUsername Thanks again for spotting this! 
The fix landed and will be available in `11.2.0-rc.1` feel free to reopen if the crash still persist
